index-scheduler/src/batch.rs

```diff
@@ -689,7 +696,67 @@ impl IndexScheduler {
 
                 Ok(tasks)
             }
+            Batch::IndexSwap { mut task } => {
+                let mut wtxn = self.env.write_txn()?;
+                let swaps = if let KindWithContent::IndexSwap { swaps } = &task.kind {
+                    swaps
+                } else {
+                    unreachable!()
+                };
+                for (lhs, rhs) in swaps {
+                    self.apply_index_swap(&mut wtxn, task.uid, lhs, rhs)?;
+                }
```

Comment:
Would it be possible for you to add a task in the checklist for checking that we don't have two times the same index in the swap list?